src/main/java/org/cryptomator/ui/keyloading/hub/AuthFlowTask.java

```diff
@@ -37,7 +39,8 @@ protected String call() throws IOException, InterruptedException {
 				.authFlow(URI.create(hubConfig.authEndpoint)) //
 				.setSuccessResponse(Response.redirect(URI.create(hubConfig.authSuccessUrl + "&device=" + authFlowContext.deviceId()))) //
 				.setErrorResponse(Response.redirect(URI.create(hubConfig.authErrorUrl + "&device=" + authFlowContext.deviceId()))) //
-				.authorize(redirectUriConsumer);
+				.authorize(HttpClient.newBuilder().connectTimeout(Duration.ofSeconds(3)).build(),
```

Comment:
Is the client's connect timeout the same as an request's timeout?

Reply:
No. Citing the above linked javadoc:
> In the case where a new connection needs to be established, if the connection cannot be established within the given duration, then [HttpClient::send](https://docs.oracle.com/en/java/javase/21/docs/api/java.net.http/java/net/http/HttpClient.html#send(java.net.http.HttpRequest,java.net.http.HttpResponse.BodyHandler)) throws an [HttpConnectTimeoutException](https://docs.oracle.com/en/java/javase/21/docs/api/java.net.http/java/net/http/HttpConnectTimeoutException.html), or [HttpClient::sendAsync](https://docs.oracle.com/en/java/javase/21/docs/api/java.net.http/java/net/http/HttpClient.html#sendAsync(java.net.http.HttpRequest,java.net.http.HttpResponse.BodyHandler)) completes exceptionally with an HttpConnectTimeoutException. If a new connection does not need to be established, for example if a connection can be reused from a previous request, then this timeout duration has no effect.

In contrast, the timeout of a HttpRequest is defined as "timespan the thread/process wait for a response". This means if we reuse an already established connection, there is effectively no timeout, because the request is encapsulated.

Reply:
[tiny-oauth2-client 0.7.0](https://github.com/coffeelibs/tiny-oauth2-client/releases/tag/0.7.0) should be available in a couple of minutes on Maven Central

Reply:
changed in 174225c